### DISCUSSION TITLE
A question about setting multiple parameters

### DISCUSSION THREAD

**Discussion by leffsu** [PatilShreyas/mutekt#6] [General]
Greetings!

Let's assume we have:

interface HomeScreenState {
val isLoading: Boolean
val title: String
val avatar: String
val items: List<Any>
val selectedItem: Any?
}

and a method like

fun update() {
_state.isLoading = true
val data = getSomeData()
_state.title = data.title
_state.avatar = data.avatar
_state.items = data.items
_state.selectedItem = data.items.firstOrNull()
}

or 

fun update() {
_state.isLoading = true
val data = getSomeData()
_state.apply {
title = data.title
avatar = data.avatar
items = data.items
selectedItem = data.items.firstOrNull()
}
}

How many times would state.collect {} block be called in a view/widget in both cases?

**Comment by PatilShreyas** [comment]
The behaviour of StateFlow is conflated. So `collect{}` block will be called only once.

If there is a delay in `getSomeData()`, then the collector will be called twice.

**Reply by aartikov** [reply]
It depends on dispatcher

**Reply by PatilShreyas** [reply]
Yes, it depends on the dispatcher. For example, if using `Main.immediate` dispatcher, then each update will cause new model creation.

**Comment by PatilShreyas** [comment]
Converting this to a discussion since it's not actually an issue.
